aspen/http/request.py

```diff
@@ -226,6 +226,13 @@ def __new__(cls, method=b'GET', uri=b'/', server_software=b'',
 
             raise Response(400, "Request is undecodable. "
                                 "(%s:%d)" % (filename, frame.f_lineno))
+
+        # set up aliases
```

Comment:
I see what you're saying. What if we customize attribute setting in the relevant places to make it an error to overwrite any of these attributes?